### ISSUE TITLE
"grunt-penthouse" doesn't see "penthouse"

### ISSUE THREAD

**Issue by alundiak** [fatso83/grunt-penthouse#16]
Installed as:
```
npm install grunt-penthouse --save-dev
```
And despite I see in grunt-penthouse `package.json` this:
```
 "dependencies": {
    "penthouse": ">= 0.5.0 <1.0.0)"
  },
```
**when I hit command `grunt penthouse` I have this message**:
> Loading "penthouse.js" tasks...ERROR
Error: Cannot find module 'penthouse'
Warning: Task "penthouse" not found. Use --force to continue.

I tried to install `penthouse` globally, no effect. Only installing locally helped.
```
npm install penthouse --save-dev
```
And only after this steps, `grunt penthouse` works as expected.

**But this shouldn't be**, @pocketjoso @fatso83  right? I mean, grunt-penthouse should see it's internal/local dependency. I shall not install it additionally. Even more, there is no mention about such movement in grunt-penthouse readme.

I think, there is some issue inside of `grunt-penthouse` (penthouse.js) when it tries to require penthouse:
```
...
var penthouse = require('penthouse');
grunt.registerMultiTask('penthouse', 'Extracts critical path CSS', function () { }
...
```

I reserached similar modules, like [`grunt-parallel`](https://github.com/iammerrick/grunt-parallel/blob/master/package.json), and I see only one difference in their `package.json`:

```
 "dependencies": {
    "q": "~0.8.12",
    "grunt": "~0.4.1",
    "lpad": "~0.1.0"
  },
```

And as result, after `npm -install --save-dev grunt-parallel` has internal `node_modules`
![image](https://cloud.githubusercontent.com/assets/2131633/12533214/ff731e92-c228-11e5-8357-7d30cd1522bf.png)

Reproduced with:
- grunt-penthouse `v.0.3.0` and  penthouse `v0.8.4`.
- grunt-cli `v0.1.13`
- grunt `v0.4.5`
- node `v4.2.4`
- npm `v2.14.12`

@pocketjoso @fatso83 is this penthouse-like issue? Or something strange with npm?

**Comment by fatso83** [comment]
Thank you for a detailed report! This makes it far easier to look into. Maybe I have made an error with the  last release. I'll look into it.

**Comment by fatso83** [comment]
I have just tried with a clean project and both version 4 and 5 of Node (and implicitly version 2 and 3 of NPM), and they all seem to work.

*My test*

Terminal 1:
```
python -m SimpleHTTPServer 9000
```

Terminal 2:
```
$ cat > Gruntfile.js << EOF
var grunt = require('grunt');
grunt.loadNpmTasks('grunt-penthouse');

grunt.initConfig({
  penthouse: {
    extract : {
        outfile : 'out.css',
        css : 'full.css',
        url : 'http://localhost:9000',
        width : 1300,
        height : 900,
        skipErrors : false // this is the default
    },
  },
});
EOF

$ touch full.css

$ grunt penthouse
Running "penthouse:extract" (penthouse) task
Done!
```

To me this seems to work quite fine, so I am closing it. Feel free to reopen it if you are unable to run the above commands successfully.

P.S. If you do decide to reopen, could you please post your `Gruntfile.js` so that I can see that everything is `require`d? Nothing is supposed to happen if you run `grunt penthouse` if you have not configured a task or loaded the `grunt-penthouse` task into the running Grunt task runner.

**Comment by alundiak** [comment]
reason? is it fixed?

**Comment by fatso83** [comment]
As stated in the comment above, I am unable to reproduce this. You can see the previous project creating a new project from scratch and running the penthouse task successfully. Please inspect that against your own project. 

As it stands, I have not seen any evidence of a bug that needs to be fixed.

**Comment by alundiak** [comment]
I do confirm, that my test prj "ph" with structure below, works fine:

<img src="https://cloud.githubusercontent.com/assets/2131633/12558110/52b3b01a-c38e-11e5-90ef-6f56ded71110.png" data-canonical-src="https://cloud.githubusercontent.com/assets/2131633/12558110/52b3b01a-c38e-11e5-90ef-6f56ded71110.png" width="200" />

This time, `grunt-penthouse` at least has `penthouse` inside internal `node_modules`.
And command `grunt penthouse` works fine.

I figured out the reason of my problem - 'grunt-penthouse' installed not correctly, or updated not correctly, or during `node-check-updates` command was something wrong, and as result grunt-penthouse lost (or did not have initially) internal `penthouse` dependency. **I dropped all `node_modules` and re-installed again. All is ok now**.

Now, topic may be 99% closed. In case of new evidence of such potential behavior, I will reopen issue.
